lib/gprd.gi

```diff
@@ -510,19 +510,29 @@ local iso;
     fi;
   fi;
 
-  # the ...Op is installed for `IsGroupHomomorphism'. So we have to enforce
-  # the filter to be set.
-  if not IsGroupHomomorphism(ghom) or not IsGroupHomomorphism(hhom) then
-    Error("mappings are not homomorphisms");
-  fi;
   return SubdirectProductOp(G,H,ghom,hhom);
 end);
 
 #############################################################################
 ##
 #M  SubdirectProduct( <G1>, <G2>, <phi1>, <phi2> )
 ##
-InstallMethod( SubdirectProductOp,"groups", true,
+
+BindGlobal("SubdirectProductFamilyPredicate",{F1, F2, F3, F4} ->
+        IsGeneralMappingFamily(F3) and IsGeneralMappingFamily(F4) and
+        IsIdenticalObj(FamilySource(F3), ElementsFamily(F1)) and 
+        IsIdenticalObj(FamilySource(F4), ElementsFamily(F2)) and 
+        IsIdenticalObj(FamilyRange(F3), FamilyRange(F4)));
+
+
+RedispatchOnCondition(SubdirectProductOp, "check mappings", 
+        SubdirectProductFamilyPredicate, 
+        [IsGroup, IsGroup, IsGeneralMapping, IsGeneralMapping],
+        [IsObject, IsObject, IsGroupHomomorphism, IsGroupHomomorphism],
+        10);
+
+   
+InstallMethod( SubdirectProductOp,"groups", SubdirectProductFamilyPredicate,
```

Comment:
Hm. So invalid arguments for `SubdirectProductOp` which this new family predicate will lead to "method not found" errors, instead of running into different (possibly more helpful) errors later on in the code (e.g. when the homomorphisms are applied and "notice" that they cannot actually be applied).

I guess one way out would be to add another method w/o family predicate, which checks if the family predicate is violated, and if so, prints a specific error message, and otherwise redispatches?